### ISSUE TITLE
Docker image arm64 support

### ISSUE THREAD

**Issue by carocad** [gogatekeeper/gatekeeper#412]
## Summary 

Add support for arm64 docker image format

## Why?

Because the current docker image has linux/amd64 arch hardcoded so it is not possible to run it as a container

## How

- use docker buildx with an appropiate target platform. See [docs](https://docs.docker.com/engine/reference/commandline/buildx_build/)
- pass the arch to the Makefile and build the go binary appropiately

## Acceptance criteria

- run without crashes on linux/arm64

## Additional Information

- I was not able to find out where is the docker image being build and pushed to the registry :/
- https://github.com/gogatekeeper/gatekeeper/issues/244

**Comment by p53** [comment]
Hi, thanks for creating issue! There image is built by quay.io on release, it builds image from Dockerfile, will look at it

**Comment by p53** [comment]
@carocad can you try 2.9.5-rc7 there is image for amd64/arm64/arm7 on quay.io

**Comment by carocad** [comment]
I will try to do this in the following days. Thanks for looking into this :)

**Comment by carocad** [comment]
@p53 I can confirm that `2.9.5-rc7` works with arm64 cpus

```bash
gatekeeper    | 2024-02-05T11:01:07.820Z        info    starting the service    {"prog": "gatekeeper", "author": "go-gatekeeper", "version": "2.9.5-rc7 (git+sha: 91a86e1, built: 03-02-2024)"}
```

**Comment by carocad** [comment]
closing this as it was solved on 2.9.5: https://github.com/gogatekeeper/gatekeeper/releases/tag/2.9.5

**Comment by p53** [comment]
Hi,

here is the release step where images are built:
https://github.com/gogatekeeper/gatekeeper/blob/03fea81dcb403ebaed8e7b740a4a0526098376c2/.github/workflows/release.yml#L103

P. Ipoth


Sent with Proton Mail secure email.
